### ISSUE TITLE
Unexpected change in number of units as detect_threshold varies

### ISSUE THREAD

**Issue by pamelland** [flatironinstitute/mountainsort5#45]
Hi,
 
Thank you for being so responsive on the posted issues; they are very helpful as we dig through our own results.
 
We have run across a surprising result while trying to determine an appropriate value for `detect_threshold` for a 2-hour recording on a single tetrode with 4 channels.
 
We began progressing iteratively using integers as candidate thresholds. When we got to 13, MS5 provided 24 units. When we tried 14, we got 2 units. Then at 15, the number of units went back up to 34.
 
We were surprised by the stark difference in number of units with 14, so we tried values closer to 14 (13.9 and 14.1). For both of these cases, the number of detected units was again on the same order as with 13 and 15.
 
To summarize the different unit counts and the number of detected events (if that helps) for the different values of `detect_threshold`:
 
13:     24 units;   ~112,000 total events
13.9:  38 units;  ~60,000 total events
14:     2 units;    ~64,000 total events
14.1:  34 units;  ~63,000 total events
15:    47 units;  ~106,000 total events

 
Do you have any instinct as to why we get such a low number of units (seemingly uniquely) when the threshold is 14? All other hyperparameters are unchanged.
 
Thanks!
 
Pake Melland

**Comment by magland** [comment]
Hi Pake,

Thanks for reporting this odd and surprising behavior. Especially strange that the number of events decreases so much for thresholds close to 14.

Did you apply bandpass filter and whitening?

I think the algorithm is not deterministic. What happens if you run it a second time with threshold 14?

Would you be willing to upload this dataset to DANDI, either as public or embargoed? If you do that then I will try to reproduce and troubleshoot on my end.

Jeremy
